### PR TITLE
Remove sleep and add ability to bulk delete from index

### PR DESCRIPTION
**Pull request by afrancis13**
Wanted to remove sleep statements from testing, and did so using the refresh method.
Also created a way for you to use the update_index function to bulk delete items. This is a quicker way than iterating through and using delete_item_index. This change does not impact the functionality of anything that previously existed in bungiesearch, and must be called with intent by the user.

### CODE REVIEW

File: bungiesearch/utils.py [hunk 1]
```diff
@@ -8,11 +8,12 @@
 from . import Bungiesearch
 
 
-def update_index(model_items, model_name, bulk_size=100, num_docs=-1, start_date=None, end_date=None):
+def update_index(model_items, model_name, action='index', bulk_size=100, num_docs=-1, start_date=None, end_date=None, commit=True):
     '''
     Updates the index for the provided model_items.
     :param model_items: a list of model_items (django Model instances, or proxy instances) which are to be indexed, or updated.
     :param model_name: doctype, which must also be the model name.
+    :param action: the action that you'd like to perform on this group of data. Must be in ('index', 'delete') and defaults to 'index'
```
Comment (ChristopherRabotin) [inline bungiesearch/utils.py]: [MInor] Could there be a period at the end of this line?

File: bungiesearch/utils.py [hunk 2]
```diff
@@ -40,20 +41,27 @@ def update_index(model_items, model_name, bulk_size=100, num_docs=-1, start_date
                     if end_date:
                         model_items = model_items.filter(**{'{}__lte'.format(index_instance.updated_field): __str_to_tzdate__(end_date)})
 
-                logging.info('Fetching number of documents to be added to {}.'.format(model.__name__))
+                logging.info('Fetching number of documents to update in {}.'.format(model.__name__))
```
Comment (ChristopherRabotin) [inline bungiesearch/utils.py]: Could this be:
`logging.info('Fetching number of documents to {} in {}.'.format(action, model.__name__))`, just like on line 47 ;)

File: bungiesearch/utils.py [hunk 3]
```diff
@@ -40,20 +41,27 @@ def update_index(model_items, model_name, bulk_size=100, num_docs=-1, start_date
...
+            logging.warning('Limiting the number of model_items to {} to {}.'.format(action, num_docs))
 
-        logging.info('Indexing {} documents on index {}.'.format(num_docs, index_name))
+        logging.info('{} {} documents on index {}'.format(action, num_docs, index_name))
         prev_step = 0
         max_docs = num_docs + bulk_size if num_docs > bulk_size else bulk_size + 1
         for next_step in range(bulk_size, max_docs, bulk_size):
-            logging.info('Indexing documents {} to {} of {} total on index {}.'.format(prev_step, next_step, num_docs, index_name))
-            bulk_index(src.get_es_instance(), [index_instance.serialize_object(doc) for doc in model_items[prev_step:next_step] if index_instance.matches_indexing_condition(doc)], index=index_name, doc_type=model.__name__, raise_on_error=True)
+            logging.info('{} documents {} to {} of {} total on index {}.'.format(action, prev_step, next_step, num_docs, index_name))
```
Comment (ChristopherRabotin) [inline bungiesearch/utils.py]: For clarity, could this message be instead?
```
logging.info('{}: documents {} to {} of {} total on index {}.'.format(action.capitalize(), prev_step, next_step, num_docs, index_name))
```

File: bungiesearch/utils.py [hunk 4]
```diff
@@ -8,11 +8,12 @@
...
-def update_index(model_items, model_name, bulk_size=100, num_docs=-1, start_date=None, end_date=None):
+def update_index(model_items, model_name, action='index', bulk_size=100, num_docs=-1, start_date=None, end_date=None, commit=True):
     '''
     Updates the index for the provided model_items.
     :param model_items: a list of model_items (django Model instances, or proxy instances) which are to be indexed, or updated.
     :param model_name: doctype, which must also be the model name.
+    :param action: the action that you'd like to perform on this group of data. Must be in ('index', 'delete') and defaults to 'index'
     :param bulk_size: bulk size for indexing. Defaults to 100.
     :param num_docs: maximum number of model_items from the provided list to be indexed.
     :param start_date: start date for indexing. Must be as YYYY-MM-DD.
```
Comment (ChristopherRabotin) [inline bungiesearch/utils.py]: Could you add a tiny description of what the `commit` parameter does (although it's obvious, it helps when reading the docs)?

File: bungiesearch/utils.py [hunk 5]
```diff
@@ -40,20 +41,27 @@ def update_index(model_items, model_name, bulk_size=100, num_docs=-1, start_date
...
             prev_step = next_step
 
-def delete_index_item(item, model_name):
+        if commit:
+            src.get_es_instance().indices.refresh(index=index_name)
+
+def delete_index_item(item, model_name, commit=True):
     '''
     Deletes an item from the index.
     :param item: must be a serializable object.
```
Comment (ChristopherRabotin) [inline bungiesearch/utils.py]: Idem here: add `:param commit: blah blah`.

**Comment by ChristopherRabotin** [comment]
Sorry I'm only getting around to review this now. Overall I like the changes, just a few things here and there because I'm quite picky. Thanks!

**Comment by afrancis13** [comment]
No problem. Thanks for the review, and let me know if you see anything else!

**Comment by ChristopherRabotin** [comment]
Awesome work @afrancis13 , thanks!

**Pull request merged**